core/cas-server-core-authentication/src/main/java/org/apereo/cas/authentication/handler/support/AbstractUsernamePasswordAuthenticationHandler.java

```diff
@@ -58,7 +58,6 @@ protected HandlerResult doAuthentication(final Credential credential) throws Gen
         }
 
         userPass.setUsername(transformedUsername);
-        userPass.setPassword(this.passwordEncoder.encode(userPass.getPassword()));
```

Comment:
It would also be very useful if you could:

- Provide a test that shows the broken behavior today.
- Provide additional tests that show how the patch fixes that error and broken behavior.